packages/x-date-pickers-pro/src/SingleInputDateRangeField/SingleInputDateRangeField.tsx

```diff
@@ -55,6 +55,25 @@ SingleInputDateRangeField.propTypes = {
   // | These PropTypes are generated from the TypeScript type definitions |
   // | To update them edit the TypeScript types and run "yarn proptypes"  |
   // ----------------------------------------------------------------------
+  /**
+   * This prop helps users to fill forms faster, especially on mobile devices.
+   * The name can be confusing, as it's more like an autofill.
+   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
+   */
+  autoComplete: PropTypes.string,
+  /**
+   * If `true`, the `input` element is focused during the first mount.
+   * @default false
+   */
+  autoFocus: PropTypes.bool,
+  className: PropTypes.string,
+  /**
+   * The color of the component.
+   * It supports both default and custom theme colors, which can be added as shown in the
+   * [palette customization guide](https://mui.com/material-ui/customization/palette/#adding-new-colors).
+   * @default 'primary'
+   */
+  color: PropTypes.oneOf(['error', 'info', 'primary', 'secondary', 'success', 'warning']),
```

Comment:
> From the DX perspective, I'm not sure that any of the TextField APIs should be duplicated here. I wonder if we shouldn't lean on this approach instead: https://mui.com/material-ui/api/menu/#inheritance. The advantage is that it makes it clear where the props are coming from, and what's the structure of the component. The downside is that people have to learn the pattern once. If they already learned it from MUI Core, they should be good.

Really good point. I did not investigate the MUI core solution to this issue deep enough.
In regards to api documentation your point makes total sense and would allow better differentiation between component and inherited props.
But it still leaves some open questions on edge cases:
- what if we inherit only some of the component props? I'd imagine it might cause a bit of confusion as to which props are actually valid
- what about PropTypes then? They may once again be way out of sync, with what props are actually available on a component. Or we don't care that much about their correctness because a possibly small amount of users are not using TS?

Reply:
> what if we inherit only some of the component props?

Then the pattern I proposed is not an option. It only works if the component forwards all the props.

> what about PropTypes then?

I think that the type should stay broad and links the source, like in this example: https://mui.com/material-ui/api/tree-item/#props `TransitionProps`.